src/rgw/rgw_op.cc

```diff
@@ -2017,16 +2017,8 @@ int RGWGetObj::handle_slo_manifest(bufferlist& bl, optional_yield y)
 
 int RGWGetObj::get_data_cb(bufferlist& bl, off_t bl_ofs, off_t bl_len)
 {
-  /* garbage collection related handling */
-  utime_t start_time = ceph_clock_now();
-  if (start_time > gc_invalidate_time) {
-    int r = store->defer_gc(s->obj_ctx, s->bucket.get(), s->object.get(), s->yield);
-    if (r < 0) {
-      ldpp_dout(this, 0) << "WARNING: could not defer gc entry for obj" << dendl;
-    }
-    gc_invalidate_time = start_time;
-    gc_invalidate_time += (s->cct->_conf->rgw_gc_obj_min_wait / 2);
-  }
+  /* garbage collection related handling:
```

Comment:
Since some version of this code will likely be resurrected, possibly controlled by a config option, I'd have preferred a "#if 0" rather than a delete for the sake of continuity between commits.

Reply:
when we're ready, we can revert the commit. i think committing and reverting is the most direct way to make a temporary transformation to the code

Reply:
really?  that doesn't match my experience, really

Reply:
@mattbenjamin @cbodley @ivancich : Is there agreement amongst everyone to revert this commit?

Reply:
@pritha-srivastava no objection, it comes out the same in the end